core/src/avm2/object.rs

```diff
@@ -153,6 +153,17 @@ pub use crate::avm2::object::xml_list_object::{
 pub use crate::avm2::object::xml_object::{xml_allocator, XmlObject, XmlObjectWeak};
 use crate::font::Font;
 
+fn borrow_data<'gc>(gc: Gc<'gc, RefLock<ScriptObjectData<'gc>>>) -> Ref<ScriptObjectData<'gc>> {
```

Comment:
why are these needed?

Reply:
I'm getting bad codegen when I remove these, though I'm not sure why, they're getting inlined. These are going to disappear as soon as ScriptObjectData gets interior mutability, though, so I don't think it's too big of a problem.